effectai/effect-network
# Add unit tests

### Context
We want to have the token contract fully testable.

### What has been done
- Re-structure the source files to make space for tests.
- Add the basic unit tests from eosio.contracts repo.
- Added CMakeFiles to compile the contract and unit tests with cmake + make.

### How to test
- Pull the docker-eos PR
- Pull this PR
- From the root of the project:
- `./scripts/restart-eos.sh && ./scripts/bootstrap.sh`
- `source ./scripts/aliases.sh`
- Follow the first block of codes from the `Compiling, deploying and testing the Effect.AI smart contract` section in the README.
- `eosio-test effect.contracts/contracts/effect.token`
- See the tests passing.
